src/domain/collaboration/callout/callout.service.ts

```diff
@@ -100,7 +103,10 @@ export class CalloutService {
     if (callout.authorization)
       await this.authorizationPolicyService.delete(callout.authorization);
 
-    return await this.calloutRepository.remove(callout as Callout);
+    const result = await this.calloutRepository.remove(callout as Callout);
+    result.id = calloutID;
```

Comment:
Doesn't remove return the id?

Reply:
No

Reply:
I now checked, I can selected it from the result and its type is `string`, not `string | undefined`, so it should be always there I would say.

Reply:
Actually this is a pattern we have widely used, so I would keep with it. Key is that the returned data has the actual ID of the entity that was removed so that the client cache can remove it.

Reply:
> I now checked, I can select it from the result and its type is string, not string | undefined, so it should be always there I would say.

It's undefined

Before remove
![image](https://user-images.githubusercontent.com/7901834/186932641-15cf2383-244e-40a8-bf59-993477b61c21.png)
After remove
![image](https://user-images.githubusercontent.com/7901834/186932677-b251390e-da40-49d5-910c-78e0729930bd.png)



>Current behaviour is by design. Missing id means entity doesn't exist in the db. And since we are mutating given entity, it's logically make sense. Depend of your use case, but you can simply do:
```ts
const removedEntityId = entity.id
await repository.remove(entity)
// now use removedEntityId where you need
```

[More on the topic](https://github.com/typeorm/typeorm/issues/7024)

Reply:
Wow, ok then it's good. Need to watch for other places we use it without reassigning.